src/mudlet.cpp

```diff
@@ -795,6 +794,12 @@ void mudlet::slot_close_profile_requested( int tab )
         }
     }
 
+    // close IRC client window if it is open.
+    if( mpIrcClientMap[pH] ) {
+        mpIrcClientMap[pH]->setAttribute(Qt::WA_DeleteOnClose);
+        mpIrcClientMap[pH]->deleteLater();
```

Comment:
Unless `mpIrcClientMap[pH]` is a `QPointer` the entry in `mpIrcClientMap` for this `pH` should be at least be set to `nullptr` here if not removed entirely. If it __is__ a `QPointer` it should set itself to `nullptr` automatically as needed when what it is pointing at "goes away"

Also if `mpIrcClientMap` IS a `QMap` then using `if( mpIrcClientMap[pH] )` WILL AUTOMATICALLY CREATE an entry in `mpIrcClientMap` - use `mpIrcClientMap.value(pH)` instead...